master/internal/api_master.go

```diff
@@ -87,3 +88,30 @@ func (a *apiServer) MasterLogs(
 		api.NewLogStoreProcessor(resp.Context(), lReq, fetch, onBatch, nil, nil),
 	).AwaitTermination()
 }
+
+func (a *apiServer) ResourceAllocationRaw(
+	_ context.Context,
```

Comment:
little late now, but in the future @dzhu may be nice to have a `QueryProtoContext` that uses `QueryxContext` underneath, especially for this one. Since it's costly, if the request died I imagine that makes sure the query gets terminated too.